platform/viewer/src/lib/localFileLoaders/dicomFileLoader.js

```diff
@@ -0,0 +1,206 @@
+import * as dcmjs from 'dcmjs';
+import cornerstoneWADOImageLoader from 'cornerstone-wado-image-loader';
+import FileLoader from './fileLoader';
+
+const DICOMFileLoader = new (class extends FileLoader {
+  fileType = 'application/dicom';
+  loadFile(file, imageId) {
+    return cornerstoneWADOImageLoader.wadouri.loadFileRequest(imageId);
+  }
+
+  getDataset(image, imageId) {
+    const dicomData = dcmjs.data.DicomMessage.readFile(image);
+    const dataset = dcmjs.data.DicomMetaDictionary.naturalizeDataset(
+      dicomData.dict
+    );
+    dataset._meta = dcmjs.data.DicomMetaDictionary.namifyDataset(
+      dicomData.meta
+    );
+
+    dataset.imageId = image.imageId || imageId;
+
+    return dataset;
+  }
+
+  getStudies(dataset, imageId) {
+    return this.datasetsToStudies([dataset]);
+  }
+
+  datasetsToStudies(datasets) {
```

Comment:
I think the PR is pretty good. My only reservation is that this function is overly complicated right now since it is only ever called on one dataset. One dataset can only produce one series and instance, so all the looping and filtering logic inside here is no longer necessary.

Reply:
ok .. thnk you